pkg/api/validation/validation.go

```diff
@@ -2844,10 +2799,16 @@ func ValidateNodeUpdate(node, oldNode *api.Node) field.ErrorList {
 	// Clear status
 	oldNode.Status = node.Status
 
+	// update taints
```

Comment:
FYI: managers code can handle toleration updates.

From usability perspective I'd say it should be possible to add Toleration, e.g. when operator is adding new taint to the cluster, and want to keep some of the Pods running.